lib/modules/manager/pep621/schema.ts

```diff
@@ -65,7 +66,11 @@ const UvSource = z.union([
 
 const UvSchema = z.object({
   'dev-dependencies': DependencyListSchema,
-  sources: LooseRecord(z.string(), UvSource).optional(),
+  sources: LooseRecord(
+    // uv applies the same normalization as for Python dependencies on sources
+    z.string().transform((source) => normalizePythonDepName(source)),
```

Comment:
Note on this: if we have 2 sources with different casing that would actually be normalised to the same name, Renovate will only keep the 2nd one. In reality, uv prevents having 2 sources that normalise to the same one anyway: https://github.com/astral-sh/uv/issues/7229.